infrastructure/cdn-in-a-box/mid/setup.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -e
+set -x
+set -m
+
+traffic_server start &
+
+INFO=$(ifconfig eth0 | grep 'inet ' | tr -s ' ')
+myIP=$(echo $INFO | cut -d' ' -f2)
+device=eth0
+gateway=$(route -n | grep $device | grep -E '^0\.0\.0\.0' | tr -s ' ' | cut -d ' ' -f2)
+mtu=$(ip addr show | grep $device | head -n 1 | cut -d ' ' -f5)
+netmask=$(echo $INFO | cut -d ' ' -f4)
+
+sed -ie "s;MY_HOSTNAME;$(hostname -s);g" /server.json
+sed -ie "s;MY_DOMAINNAME;$(dnsdomainname);g" /server.json
+sed -ie "s;MY_IFACE_NAME;$device;g" /server.json
+sed -ie "s;MY_MTU;$mtu;g" /server.json
+sed -ie "s;MY_GATEWAY;$gateway;g" /server.json
+sed -ie "s;MY_NETMASK;$netmask;g" /server.json
+sed -ie "s;MY_IP;$myIP;g" /server.json
+
+while ! nc trafficops 6443 </dev/null; do
+	echo "waiting for traffic_ops_golang:6443"
+	sleep 3
+done
+
+# Now network things. First need to authenticate
+curl -ksc cookie.jar -d '{"u":"admin","p":"twelve"}' https://trafficportal/api/1.3/user/login
```

Comment:
hard-coded user/passwd?   you need the same one as stored in traffic_ops -- use values from `variables.env`